### PR TITLE
Introducing 0.0.2 version of ocs csv

### PR DESCRIPTION
**Pull request by sacharya**
This is to get the discussion started around how we are going to do the next release after 0.0.1 version.

1. Will we use alpha/beta channels going forward? Or dev-preview/tech-preview/stable, etc? 
2. What channel would 0.0.2 be in? If it's in the same channel, there is no reason any deployment should use 0.0.1 anymore. So remove that directory altogether?

### CODE REVIEW

File: deploy/olm-catalog/ocs-operator/0.0.2/ocs-operator.v0.0.2.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,1334 @@
...
+    url: https://github.com/openshift/ocs-operator
+  maintainers:
+  - email: jarrpa@redhat.com
+    name: Jose Rivera
+  - email: kaushal@redhat.com
+    name: Kaushal M
+  maturity: alpha
+  provider:
+    name: Red Hat
+  replaces: ocs-operator.v0.0.1
```
Comment (sacharya) [inline deploy/olm-catalog/ocs-operator/0.0.2/ocs-operator.v0.0.2.clusterserviceversion.yaml]: This is really change that is required for upgrades to trigger.

**Comment by sacharya** [comment]
> I'm pretty sure we need to keep 0.0.1 and all previous csv versions. 
Right. We do need the previous versions in there but I don't think they have to be part of a channel in the package.yaml.

> We could consider something like 0.0.1 to always represent the "latest" nightly build, and 0.0.2 to always represent the dev version being tested
I am good with that to start with.

**Comment by davidvossel** [comment]
>> I'm pretty sure we need to keep 0.0.1 and all previous csv versions.

> Right. We do need the previous versions in there but I don't think they have to be part of a channel in the package.yaml.

>> We could consider something like 0.0.1 to always represent the "latest" nightly build, and 0.0.2 to always represent the dev version being tested

> I am good with that to start with.

@jarrpa @kshlm do either of you want to weigh in here? The idea would be that 0.0.1 and 0.0.2 would be special versions only used during development.

0.0.1 would point to the nightly build for whatever branch that's in question, and 0.0.2 would point to the build generated from the PR (or dev environment's) source tree. Our automation would ensure that in each PR we can update from the latest nightly to the code in the PR. This would hopefully help catch issues where we might be introducing a change that breaks upgradability.

**Comment by davidvossel** [comment]
> Right. We do need the previous versions in there but I don't think they have to be part of a channel in the package.yaml.

@sacharya the channel stuff is where this gets complicated. Here's an alternative we could consider.

We omit the 0.0.2 CSV in the nightly. So when we install a new catalog source with 0.0.2, if 0.0.1 is installed it automatically get's updated. this would avoid having multiple channels entirely. 

For the functional test we'd install the nightly, then update to a new catalog source that has 0.0.2 included.

**Comment by jarrpa** [comment]
> > > I'm pretty sure we need to keep 0.0.1 and all previous csv versions.
> 
> > Right. We do need the previous versions in there but I don't think they have to be part of a channel in the package.yaml.
> 
> > > We could consider something like 0.0.1 to always represent the "latest" nightly build, and 0.0.2 to always represent the dev version being tested
> 
> > I am good with that to start with.
> 
> @jarrpa @kshlm do either of you want to weigh in here? The idea would be that 0.0.1 and 0.0.2 would be special versions only used during development.
> 
> 0.0.1 would point to the nightly build for whatever branch that's in question, and 0.0.2 would point to the build generated from the PR (or dev environment's) source tree. Our automation would ensure that in each PR we can update from the latest nightly to the code in the PR. This would hopefully help catch issues where we might be introducing a change that breaks upgradability.

I'm okay with this. The CSV versions in the upstream are currently meaningless, it's what'll come out of the downstream builds that we'll want to handle.

**Comment by sacharya** [comment]
So it looks like one action item from this would be to create a new catalog source image with 0.0.2 that "replaces" 0.0.1.

The UPGRADE_FROM image could be nightly snapshots with 0.0.1 only.

What do you think @kshlm ?

**Comment by kshlm** [comment]
I think we should switch to proper version numbers now. We have a release-4.2 branch, on which should have CSVs with version 4.2.0. On the master branch, we should be setting the version to the next release, 4.3.0.
You would then have 4.3.0 replacing 4.2.0 in the CSV.

**Comment by sacharya** [comment]
@kshlm Updated the PR. Please review. This is targeting master, so the CSV is pointing to :latest tags. In the 4.x branches, we'd point them to their respective images.

**Comment by jarrpa** [comment]
I'd rather not see any changes to the CSV versions in this repo that don't also come with changes to the e2e tests for upgrade. I'm most in favor of David's latest suggestion.

**Comment by sacharya** [comment]
> I'd rather not see any changes to the CSV versions in this repo that don't also come with changes to the e2e tests for upgrade. I'm most in favor of David's latest suggestion.

That is exactly what this is. #189 is almost good to go which provides the e2e upgrade testing, but we need the new csv (and registry image) first. This is inline with @davidvossel's suggestion AFAIK.

**Pull request merged**